src/ecc.rs

```diff
@@ -62,6 +144,66 @@ impl CommonKeyFactory for ECCPubFactory {}
 
 impl PubKeyFactory for ECCPubFactory {}
 
+type Version = u64;
+
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+enum ECParameters<'a> {
+    // ecParametdders   ECParameters,
+    OId(asn1::ObjectIdentifier),
+    // implicitlyCA asn1::Null,
+    CurveName(asn1::PrintableString<'a>),
+}
+
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+struct ECPrivateKey<'a> {
+    version: Version,
+    private_key: &'a [u8],
```

Comment:
is this not an INTEGER in ASN.1 ?
can you add comment that references the canonical ASN.1 definition (RC or whatever doc).

Reply:
having a Cow using abstraction here like DerEncBigUint, would avoid the need for the new_owned() functions and save memcopies, and allow us to properly zeroize memory

Reply:
This comes from https://www.secg.org/sec1-v2.pdf, section C.4  which is referenced by the PKCS#11 specs. It is also written this way in "2.7 Wrapping/unwrapping private keys" of PKCS#11 3.0 to be confusing enough. I can add references or original structures to the comments if needed.

I will check the CoW. We should zeroize this one indeed.

Reply:
Ok it is an OCTET STRING, so basically we need the same code as BigUint except for the Tag which will have to identify an Octet string instead of an INTEGER.
Of course with octet string there is no leading zeros/sign problem, so that helps keeping the code simple.

Reply:
Tried to do that. I am still slightly lost here so let me know if what I did makes sense or I am off ...